docs/README.md

```diff
@@ -0,0 +1,325 @@
+a  removable media (USB stick, CDROM, etc.) it isn't typically very useful. EVE
+currently does *not* support network [PXE booting](https://en.wikipedia.org/wiki/Preboot_Execution_Environment)
+and given that Edge Nodes are typically deployed with very rudimentary networks
+attached to them it is very unlikely that EVE will support that type of
+booting. That said, it may be possible to use EVE's installer in a network boot
+scenario to install EVE on Edge Nodes when they first do a network boot in a lab
+or provisioning environment.
+
+During the installation phase, EVE assumes the target storage medium to be either
+empty or have been formatted with [GPT](https://en.wikipedia.org/wiki/GUID_Partition_Table)
+that EVE can recognize and extend. Either way, after succcesfull installation
+you will find EVE claiming the following entries int the GPT:
+
+* EFI System (vfat formatted read-only filesystem that contains GRUB trampoline)
+* IMGA (first copy of EVE's read-only root filesystem)
+* IMGB (second copy of EVE's read-only root filesystem)
+* CONFIG (vfat formatted read-only filesystem that contains Edge Node Identity specific configuration)
+* P3 (read-write filesystem that extends to the rest of the available storage medium)
+
+An astute reader may notice that the above partitioning scheme is very reminiscent
+of how [CoreOS does its paritioning](https://coreos.com/os/docs/latest/sdk-disk-partitions.html).
+We are, indeed, leveraging the good work that CoreOS folks have done so much so
+that we're using their patches for the GRUB bootloader to provide rootfs upgrade
+functionality between IMGA and IMGB root filesystems. Unlike CoreOS, though, EVE
+only relies on the partition symbolic names (and not offsets in the GPT) and hence
+can extend any existing GPT that it can recognize. This comes handy on some of
+the ARM platforms that reserve a few of the GPT entries to host things like
+firmware. Finally, it must be said that EVE doesn't support legacy partitioning
+schemes (such as MBR).
+
+EVE's root filesystem is hosted in both IMGA and IMGB partitions. This allows
+a running EVE instance to safely update itself. If you want to know more about
+how this process works read [baseimage update](BASEIMAGE-UPDATE.md) documentation.
+
+Both IMGA and IMGB host a filesystem containing the following software components
+
+* Second stage [GRUB bootloader](../pkg/grub) and [its configuration](../pkg/grub/rootfs.cfg) in /EFI folder
+* [Hypervisor](../pkg/xen) and [Linux Kernel](../pkg/kernel) in /boot folder
+* [EVE microservices](../pkg/pillar) in /containers folder
+* Minimalistic [Alpine Linux](https://alpinelinux.org)-derived UNIX environment
+
+EVE microservices are ultimately responsible for providing the entire EVE
+experience and they are described in greater details below.
+
+## Runtime Lifecycle
+
+EVE boots as any traditional operating system and it requires UEFI firmware.
```

Comment:
"EVE boots as any traditional operating system that supports UEFI."

Reply:
fixed